packages/ingest/FtpProviderClient.js

```diff
@@ -51,13 +52,31 @@ class FtpProviderClient {
   }
 
   async buildFtpClient() {
-    return new JSFtp({
-      host: this.host,
-      port: get(this.providerConfig, 'port', 21),
-      user: await this.getUsername(),
-      pass: await this.getPassword(),
-      useList: get(this.providerConfig, 'useList', false)
-    });
+    if (isNil(this.ftpClient)) {
```

Comment:
Is `buildFtpClient` somehow called multiple times on the same class instance now?

Reply:
It is called in `_list`, which is the function that is passed to `recursion` and called repeatedly to recur through the file system. We were not doing any caching of our client in this client, even though the SftpProviderClient _does_ already work this way.